bundles/binding/org.openhab.binding.gc100ir/src/main/java/org/openhab/binding/gc100ir/lib/GC100IRControlPoint.java

```diff
@@ -0,0 +1,582 @@
+            gc100ControlPoint = new GC100IRControlPoint();
+        }
+
+        return gc100ControlPoint;
+    }
+
+    private GC100IRControlPoint() {
+        deviceGCPropsMap = new LinkedHashMap<GC100IRDevice, Map<String, String>>();
+        deviceLastDisTimeMap = new HashMap<GC100IRDevice, Long>();
+        socketList = new ArrayList<GC100IRConnection>();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public boolean connectTarget(String gc100host, int module, int connector) {
+        if (gc100host == null || module <= 0 || connector <= 0) {
+            logger.warn("hostname, module or connector is invalid");
+            return false;
+        }
+
+        Set<GC100IRDevice> deviceSet = deviceLastDisTimeMap.keySet();
+
+        if ((deviceSet == null) || (deviceSet.size() == 0)) {
+            logger.warn("deviceSet is null or its size is 0.");
+            return false;
+        }
+
+        GC100IRDevice gcDevice = null;
+
+        for (GC100IRDevice gcDev : deviceSet) {
+            String ipOfDevice = gcDev.getIPAddressString();
+
+            if (ipOfDevice.equals(gc100host)) {
+                if ((module != gcDev.getModule()) || (connector != gcDev.getConnector())) {
+                    continue;
+                }
+
+                gcDevice = gcDev;
+                break;
+            }
+        }
+
+        if (gcDevice == null) {
+            logger.warn("gcDevice is null.");
+            return false;
+        }
+
+        GC100IRConnection tcpIpSoc = gcDevice.getTCPIPSocket();
+
+        if (tcpIpSoc != null) {
+            if (!tcpIpSoc.isConnected())
+                try {
+                    tcpIpSoc.connect();
+                    return true;
+                } catch (IOException e) {
+                    logger.error("IOException : ", e);
+                }
+        }
+
+        int port = 4998;
+        InetAddress ipAddress;
+
+        try {
+            ipAddress = gcDevice.getInetAddress();
+        } catch (UnknownHostException e) {
+            logger.warn("UnknownHostException : ", e);
+            return false;
+        }
+
+        String deviceType = gcDevice.getConnectorType();
+
+        if (DEVICE_TYPE_IR.equals(deviceType)) {
+            port = IR_PORT;
+        }
+
+        Socket socket = null;
+
+        try {
+            synchronized (socketList) {
+                for (GC100IRConnection tcpIpSocket : socketList) {
+                    if (tcpIpSocket == null) {
+                        continue;
+                    }
+
+                    InetAddress tcpInetAddress = tcpIpSocket.getInetAddress();
+                    int tcpPort = tcpIpSocket.getPort();
+
+                    if (tcpInetAddress.equals(ipAddress) && (tcpPort == port)) {
+                        String tcpIPString = tcpIpSocket.getInetAddress().toString();
+
+                        if (tcpIPString.contains("/")) {
+                            tcpIPString = tcpIPString.substring(1);
```

Comment:
Is this assuming that the first character is "/"?  Would `startsWith` work instead?